psycopg/psycopg/_typeinfo.py

```diff
@@ -219,20 +232,19 @@ def __init__(
         self.subtype_oid = subtype_oid
 
     @classmethod
-    def _get_info_query(
-        cls, conn: "Union[Connection[Any], AsyncConnection[Any]]"
-    ) -> str:
+    def _get_info_query(cls, conn: _AnyConn) -> str:
         if conn.info.server_version < 140000:
             raise e.NotSupportedError(
                 "multirange types are only available from PostgreSQL 14"
             )
+        # CockroachDB does not support range so no need to use _to_regtype
         return """\
 SELECT t.typname AS name, t.oid AS oid, t.typarray AS array_oid,
     t.oid::regtype::text AS regtype,
     r.rngtypid AS range_oid, r.rngsubtype AS subtype_oid
 FROM pg_type t
 JOIN pg_range r ON t.oid = r.rngmultitypid
-WHERE t.oid = %(name)s::regtype
+WHERE t.oid = to_regtype(%(name)s)
```

Comment:
I think this should be something of the likes:
```python
    @classmethod
    def _get_info_query(cls, conn: _AnyConn) -> Query:
        return sql.SQL("""`
SELECT ...
WHERE t.oid = {regtype}
""".format(regtype=cls._to_regtype(conn)))

    @classmethod
    def _to_regtype(cls, conn) -> sql.Composible:
        if ...
            return sql.SQL("to_regtype(%(name)s)")
        else:
            return sql.SQL("(%(name)s::regtype)")
```
this way we respect the `LiteralString` signature of `Query` and `sql.SQL` (Mypy doesn't check it yet, but we try to respect it already).